src/Codecs.php

```diff
@@ -227,4 +228,12 @@ public static function mixed(): Codec
     {
         return self::fromDecoder(new MixedDecoder());
     }
+
+    /**
+     * @return Codec<callable, mixed, callable>
+     */
+    public static function callable(): Codec
+    {
+        return self::fromDecoder(new CallableType());
```

Comment:
You just need to implement the `Decoder` interface, not to extend `Type`.
You did something similar with the `MixedDecoder`.